### PR TITLE
Enable linux/arm64 docker build and publish

### PR DESCRIPTION
**Pull request by dee-kryvenko**
Hey all - I am not sure if it was intentionally not enabled, but on the off chance it wasn't...

### CODE REVIEW

**Comment by deviantintegral** [comment]
I'm prototyping https://github.com/actions/actions-runner-controller by pointing it to my local colima instance on my Mac. I was surprised to see everything running through emulation and not native binaries. It looks like the underlying issue is that the runner is only amd64, so all binaries inside of it fall back to that too.

**Comment by ajschmidt8** [comment]
I hope this gets merged soon. My team is currently building our own `arm64` binaries for our use case. It would be nice if these were published from the source repository.

@nikola-jokic, @TingluoHuang, any thoughts on when/if this PR can be merged?

**Comment by nikola-jokic** [comment]
Hey everyone,

We are planning to address this issue, hopefully before the next release :relaxed: 

However, we will create a separate PR to address this issue since we also need to publish images with correct labels and have it be part of the release process

**Review by nikola-jokic — APPROVED** [review]
LGTM! @TingluoHuang I tested it on my local repo and just moved these changes there. Everything works :smile: Do you mind also taking a look just in case. I'm confident this is all we need

File: images/Dockerfile [hunk 1]
```diff
@@ -1,25 +1,30 @@
 # Source: https://github.com/dotnet/dotnet-docker
 FROM mcr.microsoft.com/dotnet/runtime-deps:6.0-jammy as build
 
+ARG TARGETOS
+ARG TARGETARCH
 ARG RUNNER_VERSION
-ARG RUNNER_ARCH="x64"
 ARG RUNNER_CONTAINER_HOOKS_VERSION=0.3.2
+ARG RUNNER_ARCH="x64"
```
Comment (TingluoHuang) [inline images/Dockerfile:8]: do we still this as an ARG?

Comment (nikola-jokic) [reply images/Dockerfile:8]: I think it should be safe to remove it
With this change, it is calculated always, and I tested it on my fork just in case

**Review by TingluoHuang — APPROVED** [review]

**Comment by wyatterp365** [comment]
<details><summary>Details</summary>
<p>
 

- [ ] **@`g😆`**



</p>
</details>

**Pull request merged**